model/API.inc.php

```diff
@@ -1324,13 +1324,13 @@ public static function getSearchParamValues($params, $param) {
 		return $sets;
 	}
 	
-	public static function getRateLimit($userID, $ip) {
+	public static function getRateLimit($params) {
 		// For individual and authorized users. One user don't need too many requests
-		if ($userID) {
+		if ($params['userID']) {
```

Comment:
Need to use `!empty($params['userID'])` here and in `getConcurrencyLimit` to avoid a strict notice.

(To address your comment, I think an object makes more sense when an unknown number of independent parameters, with no logical order, can be added later. It's possible there'll just be these two, but it seems like there might be other things to use. E.g., we might want a rate limit for all requests to a given target library, without using the userID or IP, or to a given publicly shared file in a My Publications library.)